sympy/core/function.py

```diff
@@ -2190,6 +2190,12 @@ def __new__(cls, expr, variables, point, **assumptions):
             variables = expr.variables + variables
             point = expr.point + point
             expr = expr.expr
+        elif isinstance(-expr, Subs):
```

Comment:
This now duplicates the code of the `isinstance(expr, Subs)` branch, could it be rewritten such that if a change needs to be made that it only needs to be made once rather than twice (or simply forgotten because that author is not aware of the specifics of the `-expr` branch?)

Reply:
This diff would do the job for us , what say @ThePauliPrinciple .A functional implementation looks good to me otherwise it's getting too shabby and the original committed code is looking better , not sure if this could be improved but looks fine !
```
--- a/sympy/core/function.py
+++ b/sympy/core/function.py
@@ -2185,6 +2185,21 @@ def __new__(cls, expr, variables, point, **assumptions):
         if not point:
             return sympify(expr)

+        def denest(expr, variables, point):
+            # denest
+            if isinstance(expr, Subs):
+                variables = expr.variables + variables
+                point = expr.point + point
+                expr = expr.expr
+            elif isinstance(-expr, Subs):
+                expr, variables, point = denest(-expr, variables, point)
+                expr = -expr
+            else:
+                expr = sympify(expr)
+            return expr, variables, point
+
+        expr, variables, point = denest(expr, variables, point)
         # denest
         if isinstance(expr, Subs):
             variables = expr.variables + variables
```

Reply:
It would be helpful if a comment is added why this is done, so a future author can easily understand it. In the context of the issue it is clear, but without that context it is not clear what is happening here.

I'm also wondering of `-Subs` is just a specific case of `Mul(constant, Subs(...))`, e.g. is `u(x - 3*h, y - 3*h)` currently supported? If not, could it be supported if we look for any `Mul` pattern with `Subs` rather than only `Mul(-1,Subs)`?

Reply:
Will try this one out !

>it would be helpful if a comment is added why this is done, so a future author can easily understand it. In the context of the issue it is clear, but without that context it is not clear what is happening here.
Yupp, i was planning to do the same !

Reply:
> I'm also wondering of `-Subs` is just a specific case of `Mul(constant, Subs(...))`, e.g. is `u(x - 3*h, y - 3*h)` currently supported? If not, could it be supported if we look for any `Mul` pattern with `Subs` rather than only `Mul(-1,Subs)`?

Nope , there isn't code for that currently ! cause `isinstance(-1*Mul, Subs)` is `True` if the `Mul` variable is `-Subs`, but the same isn't true for `(-constant*Mul, Subs)` . Hence master and the committed branch both give the same error !

Reply:
I think you can just generalise what you have here (but I'm not entirely certain, you'd have to confirm the result by hands):

Instead of checking for `isinstance(-expr, Subs)`, check for `isinstance(expr, Mul)`, then in that branch check if any (or more precise, exactly one) of the args is an instance of `Subs`, if it is, apply denest on the arg that is a `Subs` and multiply that result with all the other args (that should do the same if `expr == Mul(-1, Subs(..))` as the current code) but I think it handles any other constants too, including symbolic ones). If no `Subs` is found we `simplify` by default.

It might be possible to apply similar logic to e.g. `Pow` or `Function` too!

Reply:
Yupp, trying it out currently . Will try out all ways and get back to you whether we can get it done somehow , or like is it somewhat approachable with the already written code !

Reply:
> Instead of checking for `isinstance(-expr, Subs)`, check for `isinstance(expr, Mul)`, then in that branch check if any (or more precise, exactly one) of the args is an instance of `Subs`, if it is, apply denest on the arg that is a `Subs` and multiply that result with all the other args (that should do the same if `expr == Mul(-1, Subs(..))` as the current code) but I think it handles any other constants too, including symbolic ones). If no `Subs` is found we `simplify` by default.

Thanks for the suggestions , work perfectly fine ! @ThePauliPrinciple , I'll make the commit soon .